### PR TITLE
Allow cleanup(test) to run regardless of deployment

### PR DESCRIPTION
**Pull request by rtorrero**
This PR allows the cleanup playbook to be executed regardless if the deployment playbook finishes successfully or not. 

The playbooks are tested in persistent machines running in the cloud and when a deployment run doesn't finish successfully, the machine is left in an undetermined state that might affect the execution of the next CI run. This PR:
 - Separates the cleanup in a separate job to see more clearly if it executed correctly
 - Adds a continue-on-error: true to the deployment playbook so that the clean playbook still gets executed if it fails

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -123,6 +124,27 @@ jobs:
             install_method='rpm'"
       - name: Test readiness
         run: curl -k "https://${{ env.TEST_HOST_IP }}/api/readyz"
+
```
Comment (arbulu89) [inline .github/workflows/ci.yaml:127]: Was not adding `if: always()` to the step enough?

Like
```
- name: Run playbook cleanup
  if: always()
  uses: dawidd6/action-ansible-playbook@v2
  with:
```

Ref: https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f

Comment (rtorrero) [reply .github/workflows/ci.yaml:127]: The downside of using `always()` is that it literally executes always, ignoring the `needs:` (in this case, the `needs:` is only used for ordering). This means it also executes in PRs for example :/

**Comment by arbulu89** [comment]
@EMaksy @CDimonaco Now this is ready to me reviewed.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -25,9 +25,9 @@ jobs:
 
   build-and-push-container-images:
     name: Build and push container images
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-20.04
```
Comment (EMaksy) [inline .github/workflows/ci.yaml:28]: Why the change to a lower version ?
Ubuntu 22 to Ubuntu 20 ? 
Something python related?

**Review by EMaksy — APPROVED** [review]
LGTM, just left a question :)

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (arbulu89) [reply .github/workflows/ci.yaml:28]: Yes.
https://github.com/trento-project/web/pull/2713

**Review by CDimonaco — APPROVED** [review]
